Loopback mode documentation (#1)
abhilashendurthi/testvectors: README.md

* `target.pb.txt` stores the IP and port of the switch under test which could be used by a Test Vectors runner to connect to the switch. E.g. `address: "127.0.0.1:28000"`
* `portmap.pb.txt` contains a list of entries, each of which stores infomation related to a specific switch port used in the Test Vectors including port number, port type (see proto/portmap/portmap.proto for definition) and name of physical or virtual interface connected to the switch port. E.g. a portmap entry with `port_number: 58`, `interface_name: "ens6f0"` and `port_type: 0` means interface `ens6f0` is connected to switch port `58` and could be used as both ingress and egress to switch. The interfaces could be used by a Test Vectors runner to send or receive packets when Test Vectors contain Actions/Expectations that involve packets sending/receiving in the data plane.
> Note: It is recommended to mark ports used in Test Vectors as either ingress or egress because it is not supported in the loopback mode to use a port that serves as both ingress and egress to the switch. See more details below.

### Loopback Mode

> In order to support running Test Vectors in [loopback mode](https://github.com/stratum/testvectors-runner/blob/master/docs/loopback.md), it is required to specify each port used in Test Vectors as either ingress or egress to the switch. In addition, extra `Insert*` and `Delete*` Test Vectors are provided as part of loopback mode setup and teardown.

Reviewer: ```suggestion
In order to support running Test Vectors in [loopback mode](https://github.com/stratum/testvectors-runner/blob/master/docs/loopback.md), each port used in Test Vectors should be specified as either ingress or egress to the switch. In addition, extra `Insert*` and `Delete*` Test Vectors are provided as part of loopback mode setup and teardown.
```
